### ISSUE TITLE
[BUG] Issues on arm64

### ISSUE THREAD

**Issue by flanter21** [waydroid/waydroid#1387]
### Describe the bug

I've been trying for months now to try to get Waydroid to run in a linux virtual machine on my arm64 mac. I've tried ubuntu 23.10 and 24.04 and fedora 39 and fedora 40, but none of them would run always leaving some variation of an ominous

```
[gbinder] Service manager /dev/binder has appeared
[gbinder] WARNING: Service manager /dev/binder has died
```
I even tried running them in both vmware and utm and tweaking VM settings to no avail.

I tried previous solutions that had been found like installing python3-gbinder (from the waydroid sid repo) or [anbox-modules](https://github.com/choff/anbox-modules). Eventually I gave up thinking it might be fixed in the next version, even if it didn't at the time work in either fedora or ubuntu. I realised eventually that the stock kernel on ubuntu doesn't have binder support so I compiled a new kernel with it, however, the same issue still occurred, though much less frequently, mostly replaced by `WayDroid container is not listening`. Note fedora does come with binder on arm64.

While I would get resource busy when installing anbox-binder on fedora, on ubuntu I would get a compile issue. I've attached it as [make.log](https://github.com/waydroid/waydroid/files/15151875/make.log). It's weird because there were no errors during compilation but no binary was produced.

I'm really not sure where to go from here. Any ideas would be welcome.

### Waydroid version

1.4.2

### Device

Linux Desktop

### Operating System

Ubuntu 24.04 arm64

### Kernel version

6.8.7

### Desktop Environment

GNOME 46

### GPU

N/A

### Logs

It doesn't run but here they are anyways.

[/var/lib/waydroid/waydroid_base.prop](https://github.com/waydroid/waydroid/files/15152079/waydroid_base.prop.txt)
[/var/lib/waydroid/waydroid.cfg](https://github.com/waydroid/waydroid/files/15152080/waydroid.cfg.txt)
[/var/lib/waydroid/waydroid.prop](https://github.com/waydroid/waydroid/files/15152082/waydroid.prop.txt)
[/var/lib/waydroid/waydroid.log](https://github.com/waydroid/waydroid/files/15152095/waydroid.log)
`sudo waydroid shell -- logcat -d | tee logcat.txt` - `logcat: Logcat read failure: No such file or directory`
`sudo dmesg | tee dmesg.txt` - [dmesg.txt](https://github.com/waydroid/waydroid/files/15152059/dmesg.txt)
[make.log of trying to compile anbox-binder.log](https://github.com/waydroid/waydroid/files/15152108/make.log)

**Comment by cody8764** [comment]
Did you edit the config file for Waydroid to use arm64 instead of x86_64?

**Comment by flanter21** [comment]
> Did you edit the config file for Waydroid to use arm64 instead of x86_64?

No but I believe it was already downloading the arm64 image anyways.

**Comment by andmpel** [comment]
Yeah I'm having the same issue. Tried Ubuntu, Debian, Fedora, all arm64 distros. Same errors received by OP.

**Comment by flanter21** [comment]
> Yeah I'm having the same issue. Tried Ubuntu, Debian, Fedora, all arm64 distros. Same errors received by OP.

I tried again just in the past few days and sadly haven't got there. I've checked that I have binder installed, even recompiling the kernel just in case [this](https://github.com/waydroid/waydroid/issues/136#issuecomment-1885022413) might've been the cause of the issue. psi was also definitely on, as my kernel boot options included CONFIG_PSI=y and i could access /proc/pressure, with it updating in real time.

I tried running as sudo with weston and pulseaudio, just in case it was a permission issue. I likewise tried with [SELinux disabled](https://docs.fedoraproject.org/en-US/quick-docs/selinux-changing-states-and-modes/#_changing_selinux_modes_at_boot_time) to no avail.

I also tried running the component parts separately (eg. mounting the different files manually as waydroid had trouble with that). It ended up not really achieving much but I found out something weird: Waydroid needs the directories where it wants to bind the folders to have already been created as it will not create folders in the rootfs by itself (eg. dev mnt_extra tmp var run). So once I did that, it didn't have issues with binding anymore. I noticed that I didn't run into this issue on ubuntu.

I ended up finding out from my journalctl that it's either because [cgroups aren't permitted](https://github.com/waydroid/waydroid/issues/420) or it [won't run 32-bit elf1s](https://github.com/waydroid/waydroid/issues/1070) (both at the same time). I tried qemu-user for a long time but just didn't get it to work, but I eventually fixed cgroups (by chowning the rootfs).

Fedora issues that were fixed
1) Not creating folders to mount tmpfs - can just make them, then they'll mount normally next time.
2) Network error - [fix](https://fedoraproject.org/wiki/LXC#Networking) by `systemctl enable --now lxc-net.service`
3) Not sure, but gbinder seems to not work??

Ubuntu issues that were fixed
1) Namespaces not being made - sudo chown -R 0:0 /var/lib/waydroid/rootfs

So that's kind of where I'm stuck. I didn't really get much further despite disabling selinux and apparmor. Redroid works ig.

**Comment by Squishy123** [comment]
still having this issue
